mantidproject/mantid
# Add option to use chisq to fit diff constants PDCalibration

**Description of work.**

Have added option to use chisq as cost function when fitting the diffractometer constants in PDCalibration.
During testing I noticed that the stability of the fits could be improved by initialising BackToBackExponential coefficients from the Parameters.xml file for each spectra rather than taking A and B from the previous 'good' fit.

**To test:**

(1) Run the below script without error

```
Load(Filename=r'\\isis.cclrc.ac.uk\inst$\ndxenginx\instrument\data\cycle_12_4\ENGINX00193749.nxs', 
     OutputWorkspace='193749')

dvalues = (2.705702376, 1.91322,1.6316,1.56214,1.35285,1.24146, 
    1.21003,1.1046,0.95661,0.855618)
               
PDCalibration(InputWorkspace='193749',
               TofBinning=[10000,-0.0005,46000],
               PeakPositions=dvalues,
               PeakWindow = 0.05, # d-spacing around which to look for peak
               MinimumPeakHeight = 1,
               PeakFunction = 'Gaussian',
               CalibrationParameters =  'DIFC+TZERO',
               OutputCalibrationTable='cal',
               DiagnosticWorkspaces = 'diag',
               UseChiSq = True)
               
```
(2) Check that when `UseChiSq = True` the Gaussian fit parameters feature in the` _fitparam` table in the `_diag `group of workspaces (i.e. instead of the width there is a column called sigma). There should also be an additional table called `_fiterror` which contains the error on the fitted parameters which is not created when `UseChiSq = False`.

Fixes #30287




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
